NOTES.md

```diff
@@ -1,3 +1,191 @@
+    * `changes` can now be placed after `between` and `filter` (#3232)
+    * `changes` can also be used on a few other commands for which there should be a comprehensive list here
+* Server
+  * Made buffered IO the default, added a `--direct-io` flag to enable direct IO and deprecated the `--no-direct-io` flag (#3391)
+* Python driver
+  * `rethinkdb export` now exports secondary index information and `rethinkdb import` re-creates those indexes unless it is given the `--no-secondary-indexes` flag (#3484)
+
+## Improvements ##
+
+* Web UI
+  * The web assets are now compiled into the `rethinkdb` executable (#1093)
+  * `getAll` queries on secondary indexes are now shown in the performance graph (#2379)
+  * Added `getField` to the autocompletion in the data explorer (#2624)
+  * Added support for changefeeds in the data explorer (#2643)
+  * Reduced the amount of data used and computations performed by the dashboard (#2786)
+  * Swapped the order of the "Load" and "Remove" buttons (#3414)
+  * The inaccessible database page was removed (#3491)
+  * Added a secondary index monitor to the dashboard (#3493)
+* Server
+  * Removed code used to support outdated tests (#1524)
+  * Improved implementation of erase range operations (#2034)
+  * Now use `jemalloc` by default instead of `tcmalloc`, which solves some apparent memory leaks (#2279)
+  * Replaced vector clocks with naïve timestamp-based conflict resolution (#2784)
+  * No longer complain when `stderr` can't be flushed (#2790)
+  * Replaced uses of the word "machine" to use "server" instead. The old `--machine-name` flag is now `--server-name` (#2877, #3254)
+  * The server now prints its own name on startup (#2992)
+  * Adjusted the formatting of log levels and no longer print `info:` lines to stderr (#3040)
+  * Added a `--no-default-bind` option that prevents the server from listening on loopback addresses (#3154)
+  * Lower the CPU load of an idle server (#3185)
+  * Now ignore the OS disk cache when calculating available memory (#3193)
+  * Now use `kqueue` instead of `poll` for events on OS X (#3403)
+  * Tables now become available faster and table metadata takes less space on disk (#3463)
+  * Queries that perform multiple HTTP requests now share the same cookies (#3469)
+  * Made resolving of ties in secondary indexes more consistent (#3550)
+  * New version notifications are added to the server logs (#3170)
+* Testing
+  * The ReQL tests can now run in parallel (#2305, #2672)
+  * The Ruby driver is now tested against different versions of Ruby (#2526)
+  * Relaxing floating point comparison to account for rounding issues (#2912)
+  * No longer depend on external HTTP servers (#3047)
+  * Switched to using the new cluster management features in the test suite (#3398, #3401)
+* ReQL
+  * Allow creating evenly-spaced shards (#1679)
+  * Allow querying the version of the server (#2698)
+  * Changefeeds are no longer aborted when the primary replica changes (#2757)
```

Comment:
Actually the issue was about making sure *that* the changefeeds are aborted, since we are not handling this case correctly.
What we did was simply to make sure that this is indeed happening, and we do still abort changefeeds when switching the primary.

From Karl: "The current behavior is that all changefeeds are ended on a table when the shards change on that table. There does not seem to be any regard to whether it needs to end or not."